crates/web-sys/src/features/gen_MapLikeIterator.rs

```diff
@@ -0,0 +1,22 @@
+#![allow(unused_imports)]
+use super::*;
+use wasm_bindgen::prelude::*;
+#[wasm_bindgen]
+extern "C" {
+    # [wasm_bindgen (is_type_of = | _ | false , extends = :: js_sys :: Object , js_name = MapLikeIterator , typescript_type = "MapLikeIterator")]
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    #[doc = "The `MapLikeIterator` class."]
+    #[doc = ""]
+    #[doc = "[MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/API/MapLikeIterator)"]
+    #[doc = ""]
+    #[doc = "*This API requires the following crate features to be activated: `MapLikeIterator`*"]
+    pub type MapLikeIterator;
```

Comment:
i was thinking maybe here the simplest solution for now would be to return a `js_sys::Iterator` directly (this would also circumvent the need for some way to omit the MDN link because we wouldn't have a dedicated `MapLikeIterator`).

while `js_sys::Iterator` isn't best fit (it *does* return a `Result` on `next()`), we would get all its benefits, like `js_sys::Iter` compatibility, automatically.

neither of these solutions is ideal though (each `maplike` could have its own (generic?) iterator type, since the types for its keys and values is known). however, i agree with you that not having an `Iterator` implementation is pretty inconvenient.

Reply:
i will return `js_sys::Iterator` for simplicity here